### PR TITLE
chore: bump cosi-project/runtime

### PR DESCRIPTION
**Pull request by DmitriyMV**
And update all `ResourceDefinition` docs and type names. Drop unused functions and names.

### CODE REVIEW

**Comment by DmitriyMV** [comment]
Minor additions to #6849

File: go.mod [hunk 1]
```diff
@@ -40,7 +40,7 @@ require (
 	github.com/containernetworking/plugins v1.2.0
 	github.com/coreos/go-iptables v0.6.0
 	github.com/coreos/go-semver v0.3.1
-	github.com/cosi-project/runtime v0.3.0-alpha.5
+	github.com/cosi-project/runtime v0.3.0-alpha.5.0.20230214191708-e6466f3cfde4
```
Comment (smira) [inline go.mod:43]: bump in `pkg/machinery` as well?

**Review by smira — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (DmitriyMV) [reply go.mod:43]: Done!

**Pull request merged**